packages/typegpu/tsup.config.ts

```diff
@@ -11,7 +11,7 @@ export default defineConfig({
   outDir: 'dist',
   format: ['cjs', 'esm'],
   tsconfig: './tsconfig.json',
-  target: 'es2017',
+  target: 'es2022',
```

Comment:
Switching to `es2022` seems to improve performance of vector creation. I'm not sure why yet, but given WebGPU did not exist in es2017, we might as well switch it.

Reply:
Will remove from this PR as it is kinda unrelated